hibernate-core/src/main/java/org/hibernate/dialect/Database.java

```diff
@@ -501,16 +501,20 @@ public Dialect resolveDialect(DialectResolutionInfo info) {
 						return new SQLServer2008Dialect();
 					}
 					case 11:
-					case 12:
-					case 13: {
+					case 12: {
 						return new SQLServer2012Dialect();
 					}
+					case 13:
+					case 14:
+					case 15: {
+						return new SQLServer2016Dialect();
```

Comment:
is `case 16:` intentionally missing?

I guess it works via the fallback on `latestDialectInstance` but it doesn't seem very intuitive

Reply:
AFAIK version 2019 corresponds to 15.x and there is no newer version released yet. But if you think it makes sense, I could of course also add `16` here.

Reply:
ah right, ok I forgot that the major version doesn't match the "product" version name. All good then, thanks!